providers/implementations/kdfs/argon2.c

```diff
@@ -0,0 +1,1488 @@
+/*
+ * Copyright 2017-2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <limits.h>
+#include <string.h>
+#include <crypto/evp.h>
+#include <openssl/evp.h>
+#include <openssl/objects.h>
+#include <openssl/crypto.h>
+#include <openssl/kdf.h>
+#include <openssl/err.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include <internal/worker.h>
+#include <internal/numbers.h>
+#include <prov/implementations.h>
+#include <prov/provider_ctx.h>
+#include <prov/providercommonerr.h>
+#include <prov/blake2.h>
+
+#ifndef OPENSSL_NO_ARGON2
+
+# define ARGON2_MIN_LANES UINT32_C(1)
```

Comment:
Is UINT32_C c99? (That might be a problem).

Reply:
To my knowledge, yes (https://busybox.net/~landley/c99-draft.html#B.17).